design/operator_design.md

```diff
@@ -0,0 +1,129 @@
+# YugabyteDB Operator Design
+
+A cluster of YugaByte DB should be created using the `ybcluster.yugabytedb.yugabyte.com` custom resource definition. Below is a sample custom resource spec for creating a 3-master, 3 t-servers cluster using the CRD. The sample is followed by an explanation of different configuration options available on the YugaByte DB CRD.
+
+
+```yaml
+apiVersion: yugabytedb.yugabyte.com/v1alpha1
+kind: YBCluster
+metadata:
+  name: yugabytedb
+  namespace: yugabytedb
+spec:
+  image:
+    repository: yugabytedb/yugabyte     # Default.
+    tag: master                         # Default.
+    pullPolicy: IfNotPresent            # Default
+  tls:
+    enabled: false
+    # rootCA:
```

Comment:
I'm unfamiliar with defining operators; do these have to be commented out even if `tls.enabled: false`? e.g. couldn't `rootCA.cert/key` be left uncommented, but left empty, as long as `tls.enabled: false`?

Reply:
That's right. User will have to either give values for `cert/key` or omit/comment them. The values can't be null.
It's not because of how operators need to be defined but rather, the Open API V3 schema validations available for K8s CRDs. K8s CRDs validation schema doesn't support some keywords from Open API V3, including `nullable`.